### PR TITLE
BAEL-5290 - java_operators_tutorial

### PR DESCRIPTION
**Pull request by silviorazec**
This is the code of the article about java operators.

### CODE REVIEW

**Review by JavierPeris — CHANGES_REQUESTED** [review]
Please review the PR title. It must start with the Jira id.

**Comment by JavierPeris** [comment]
We should get rid of the maven wrapper and gitignore they are not needed.

File: java-operators/test/com/baeldung/operators/AssignmentTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		number1 = 10;
+		number1 = number1 % 2;
+		assertEquals(number1, 0);
+		
+		number2 = 10;
+		number2 %= 2;
+		assertEquals(number1, 0);
+		
+		
+	}
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/operators/AssignmentTest.java:56]: Could you move each operation to a new test? I think it would be more readable.

**Review by silviorazec — COMMENTED** [review]
I have deleted it.

**Review by silviorazec — COMMENTED** [review]
I have changed the class of the test.

**Comment by JavierPeris** [comment]
@silviorazec they are still a couple of files related to the Maven wrapper: 

- java-operators/mvnw
- java-operators/mvnw.cmd

Could you remove them?

File: java-operators/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (JavierPeris) [inline java-operators/src/main/resources/application.properties:1]: I think we could remove this file. There's no need of properties for this topic.

File: java-operators/test/com/baeldung/operators/ArithmeticTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class ArithmeticTest {
+
+	
+	@Test
+	public void arithmeticAddTiveTest() {
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/operators/ArithmeticTest.java:11]: There's a typo in the test name.

File: java-operators/test/com/baeldung/operators/EqualityAndRealationalTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class EqualityAndRealationalTest {
+
+	
+	@Test
+	public void greaterThanorEqualToTest() {
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/operators/EqualityAndRealationalTest.java:11]: The case should be: **greaterThanOrEqualToTest**.

File: java-operators/test/com/baeldung/operators/TypeComparisonTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		
+		Object car = new Car();
+		assertEquals(car instanceof Airplane, false);
+		
+		Object airplane = new Airplane();
+		assertEquals(airplane instanceof Car, false);
+	}
+	
+	@Test
+	public void carAndAirplaneAreTransporTest() {
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/operators/TypeComparisonTest.java:24]: There's a typo in the Transport word.

File: java-operators/test/com/baeldung/operators/TypeComparisonTest.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		
+		Car car = new Car();
+		assertEquals(car instanceof Transport, true);
+		
+		Airplane airplane = new Airplane();
+		assertEquals(airplane instanceof Transport, true);
+	}
+	
+	@Test
+	public void transporCanBeCarOrAirplaneTest() {
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/operators/TypeComparisonTest.java:34]: There's a typo in the Transport word.

File: java-operators/test/com/baeldung/operators/OtherAssignmentTest.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+		assertEquals(number1, 5);
+		
+		int number2 = 10;
+		number2 /= 2;
+		assertEquals(number2, 5);
+		
+	}
+	
+	@Test
+	public void assignmentRemaiderTest() {
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/operators/OtherAssignmentTest.java:57]: There´s a typo in the remainder word.

File: java-operators/src/main/resources/application.properties (hunk 2 above)
Comment (silviorazec) [reply java-operators/src/main/resources/application.properties]: done

File: java-operators/test/com/baeldung/operators/ArithmeticTest.java (hunk 3 above)
Comment (silviorazec) [reply java-operators/test/com/baeldung/operators/ArithmeticTest.java]: done

File: java-operators/test/com/baeldung/operators/EqualityAndRealationalTest.java (hunk 4 above)
Comment (silviorazec) [reply java-operators/test/com/baeldung/operators/EqualityAndRealationalTest.java:11]: done

File: java-operators/test/com/baeldung/operators/TypeComparisonTest.java (hunk 5 above)
Comment (silviorazec) [reply java-operators/test/com/baeldung/operators/TypeComparisonTest.java]: done

File: java-operators/test/com/baeldung/operators/TypeComparisonTest.java (hunk 6 above)
Comment (silviorazec) [reply java-operators/test/com/baeldung/operators/TypeComparisonTest.java]: done

File: java-operators/test/com/baeldung/operators/OtherAssignmentTest.java (hunk 7 above)
Comment (silviorazec) [reply java-operators/test/com/baeldung/operators/OtherAssignmentTest.java]: done

File: java-operators/test/com/baeldung/operators/ArithmeticTest.java (hunk 3 above)
Comment (JavierPeris) [reply java-operators/test/com/baeldung/operators/ArithmeticTest.java]: The correct English name is addition or sum.

File: java-operators/HELP.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# Getting Started
+
+### Reference Documentation
+For further reference, please consider the following sections:
+
+* [Official Apache Maven documentation](https://maven.apache.org/guides/index.html)
+* [Spring Boot Maven Plugin Reference Guide](https://docs.spring.io/spring-boot/docs/2.6.1/maven-plugin/reference/html/)
+* [Create an OCI image](https://docs.spring.io/spring-boot/docs/2.6.1/maven-plugin/reference/html/#build-image)
```
Comment (JavierPeris) [inline java-operators/HELP.md:8]: These references are not related to your article. Please have a look at other README files and what kind of links are usually referenced.

File: java-operators/pom.xml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.6.1</version>
+		<relativePath/> <!-- lookup parent from repository -->
```
Comment (JavierPeris) [inline java-operators/pom.xml:9]: Review the parent path, you don´t need the spring-boot-starter, the parent-java baeldung artifact should be used instead.

File: java-operators/pom.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+	</parent>
+	<groupId>com.baeldung.operators</groupId>
+	<artifactId>operator-java</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>operator-java</name>
+	<description>A short introduction to Java Operators</description>
+	
+	<properties>
+	    <maven.compiler.target>10</maven.compiler.target>
+	    <maven.compiler.source>10</maven.compiler.source>
```
Comment (JavierPeris) [inline java-operators/pom.xml:19]: If you don´t need any Java 10 feature, please use as default Java 8.

**Comment by JavierPeris** [comment]
Please reformat your code using the Baeldung Java formatter.

File: java-operators/test/com/baeldung/java/operators/UnaryTest.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+		assertEquals(number, 2);
+		
+		number++;
+		
+		assertEquals(number, 3);
+		
+	}
+	
+	@Test
+	public void whenUseTheOperatorIncrementAndDecrementInTheMoment() {
```
Comment (JavierPeris) [inline java-operators/test/com/baeldung/java/operators/UnaryTest.java:40]: Watch out when you use the And word in a test, it can give you a hint that you can split the test into two tests.

**Comment by JavierPeris** [comment]
You should not create a new module for your code if an already suitable module is available. In this case, I think your code can fit into tutorials/core-java-modules/core-java-lang-operators/
